crates/cxx-qt-build/src/lib.rs

```diff
@@ -623,6 +767,20 @@ impl CxxQtBuilder {
             }
         }
 
+        // Build inits which are generated
+        self.build_inits_for_qrc();
+        self.build_inits_for_qml_plugins();
+        if qtbuild.version().major < 6 {
+            // Inject Qt 5 compat type registers
+            self.inits.push(CxxQtBuilderInit::new(
+                "cxx_qt_qt5_compat",
+                include_str!("std_types_qt5.cpp"),
+            ));
+        }
+
+        // Generate init method
+        cc_builder_whole_archive.file(self.write_init_method());
```

Comment:
appears we still need the `+whole-archive` internally though? Unless is there a way of removing this too :thinking:

Reply:
it might only be Cargo builds that now need it? As seems they can't tell that the extern's from Rust are linking through to the init method.

Reply:
I think so, but this also means we don't need the `init` method at all in our cargo-only builds, right?
Initialization can just work "as-is" in that case, as we don't have a problem there iirc.

Reply:
Yes it means we don't need the `init` currently in Cargo, although it'd be nice to remove this whole-archive and have the linker figure things out.

However i wonder if we still document that you should use it, as later it gives us a place to register/setup anything globally. Eg what if we wanted to setup Qt logging or something?